.github/workflows/lint.yml

```diff
@@ -1,50 +1,54 @@
-name: Linters
+# lint runs all linters in this repository. This workflow is run on every pull
+# request and push to main.
+name: lint
 on:
   push:
     tags:
       - v*
     branches:
       - main
   pull_request:
+  workflow_call:
 
 jobs:
-  golangci:
-    name: lint
+  golangci-lint:
+    name: golangci-lint
     runs-on: ubuntu-latest
+    timeout-minutes: 8
     steps:
-      - uses: actions/checkout@v3
       - uses: actions/setup-go@v4
         with:
           go-version: 1.19
-      - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.4.0
+      - uses: actions/checkout@v3
+      - uses: technote-space/get-diff-action@v6.1.2
         with:
-          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
-          version: latest
-
-          # Optional: working directory, useful for monorepos
-          # working-directory: somedir
-
-          # Optional: golangci-lint command line arguments.
-          # args: --issues-exit-code=0
-
-          # Optional: show only new issues if it's a pull request. The default value is `false`.
-          # only-new-issues: true
+          # This job will pass without running if go.mod, go.sum, and *.go
+          # wasn't modified.
+          PATTERNS: |
+            **/**.go
+            go.mod
+            go.sum
+      - uses: golangci/golangci-lint-action@v3.4.0
+        with:
+          version: v1.50.1
+          args: --timeout 10m
+          github-token: ${{ secrets.github_token }}
+        if: env.GIT_DIFF
```

Comment:
where is this getting defined? I don't see an `env` section nor a statement about secrets being inherited.

Reply:
Is it set by `get-diff-action`?

Reply:
https://github.com/technote-space/get-diff-action/tree/v6.1.2/

The `technote-space/get-diff-action` set's the `GIT_DIFF` environment in the current job so it can be used by another step.